### ISSUE TITLE
QuEST does not compile with GCC 9.x

### ISSUE THREAD

**Issue by fcooper8472** [QuEST-Kit/QuEST#173]
Following the quick start in the README, I get the following error with GCC 9.1.0 (but no errors with GCC 8.x or 7.x).

CMake output:
```
-- The C compiler identification is GNU 9.1.0
-- The CXX compiler identification is GNU 9.1.0
-- Check for working C compiler: /usr/bin/gcc-9
-- Check for working C compiler: /usr/bin/gcc-9 -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/g++-9
-- Check for working CXX compiler: /usr/bin/g++-9 -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- GPU acceleration is OFF
-- OMP acceleration is ON
-- MPI distribution is OFF
-- Found OpenMP_C: -fopenmp (found version "4.5") 
-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Found OpenMP: TRUE (found version "4.5")  
-- Compiling tutorial_example.c to executable demo
-- Found PythonInterp: /usr/local/bin/python3.4 (found version "3.4.9") 
-- Testing is ON
-- Configuring done
-- Generating done
-- Build files have been written to: /home/fergus/GitRepos/QuEST/build
```
Compilation error (snippet):
```
[ 10%] Building C object QuEST/CMakeFiles/QuEST.dir/src/CPU/QuEST_cpu.c.o
/home/fergus/GitRepos/QuEST/QuEST/src/CPU/QuEST_cpu.c: In function ‘densmatr_oneQubitDegradeOffDiagonal’:
/home/fergus/GitRepos/QuEST/QuEST/src/CPU/QuEST_cpu.c:53:10: error: ‘numTasks’ not specified in enclosing ‘parallel’
   53 | # pragma omp for schedule (static)
      |          ^~~
/home/fergus/GitRepos/QuEST/QuEST/src/CPU/QuEST_cpu.c:46:10: error: enclosing ‘parallel’
   46 | # pragma omp parallel \
      |          ^~~
```

**Comment by TysonRayJones** [comment]
That's weird: `numTasks` is declared `const`!

```
const long long int numTasks = qureg.numAmpsPerChunk;
```

I understand in OpenMP 4.5, const variables were automatically shared. For some compilers, making it explicitly shared [doesn't compile](http://jakascorner.com/blog/2016/07/omp-default-none-and-const.html)!

A disappointing solution is to remove `const` in all declarations of vars used in OpenMP blocks and add them explicitly to `shared`, but perhaps there's something else going on/wrong

**Comment by TysonRayJones** [comment]
patched in ec9871bd57fe8c859bd68a1a5c284b2b7e9ffee7
